analytics/models.py

```diff
@@ -61,39 +40,85 @@ def save_image(instance, filename):
         img_jpeg.save(temp_handle_jpeg, 'JPEG')
         temp_handle_jpeg.seek(0)
         # filename
-        filename, ext = os.path.splitext(instance.image_name)
-        s3key_jpg, ext = os.path.splitext(instance.s3key)
-        suf = SimpleUploadedFile(instance.image_name, temp_handle.read(), content_type='image/tif')
-        suf_jpg = SimpleUploadedFile(filename + '.jpg', temp_handle_jpeg.read(), content_type='image/jpeg')
+        suf = SimpleUploadedFile(os.path.basename(s3key), temp_handle.read(), content_type='image/tif')
+        suf_jpg = SimpleUploadedFile(os.path.basename(s3key_jpg) + '.jpg', temp_handle_jpeg.read(), content_type='image/jpeg')
         # instance.image_orig.save(instance.s3key, suf, save=False)
         temp_handle.seek(0)
-        instance.image_orig_sha1 = hashlib.sha1(temp_handle.read()).hexdigest()
-        instance.image_jpeg.save(s3key_jpg + '.jpg', suf_jpg, save=False)
-    except ValueError:
-        print("Something went wrong with LungmapImage column image_orig upload_to function save_image.")
-    upload_dir = os.path.join('image', instance.s3key, instance.image_name)
-    return upload_dir
+        image_orig_sha1 = hashlib.sha1(temp_handle.read()).hexdigest()
+        return suf, image_orig_sha1, suf_jpg
+    except ValueError as e:
+        raise e
+
 
 
+class Experiment(models.Model):
+    experiment_id = models.CharField(max_length=25, primary_key=True)
```

Comment:
We did agree to use the auto incrementing primary keys (as opposed to the Lungmap "primary keys"). I switched to auto incrementing for all other tables, but kept this table's primary key as `encounter_id` simply because it makes linking back to Lungmap easier. I also like that the endpoints contain the lungmap id so that if things need to be qc'd against lungmap it's easy to do so just by looking at the endpoint. We can change, but I thought I'd see if I could convince you otherwise first?